replace schemamama with sqlx migrations

syphar opened the pull request:
This drops our schemamama migrations and replaces them with sqlx migrations, providing an initial migration file that represents just the current database schema. 

Just running `database migrate` will also migrate the migration state, so updates should be seamless. 

I removed running the reverse migrations from the teardown for every test case, I assume it's good enough to clean the whole schema. I added a separate check in CI to test the full setup & teardown only once. 

There is one case where this setup would break: when a developer has set `DATABASE_URL` and wants to run `cargo run -- database migrate` it won't compile. `sqlx migrate run` should work then. I wonder if we should update the documentation to use `sqlx migrate run`, though this would mean that we need to install the sqlx CLI in more places. 

Also I would love some feedback on what documentation should be updated?

Nemo157 commented:
> I removed running the reverse migrations from the teardown for every test case, I assume it's good enough to clean the whole schema. I added a separate check in CI to test the full setup & teardown only once.

Running the down migrations after each test was added to try and get a variety of data in the database that might cause issues in the down migration logic. Otherwise we're only really checking the down migration syntax, not what it does to data.

Nemo157 commented on src/db/mod.rs:
```diff
@@ -16,7 +17,58 @@ mod add_package;
+    .await?
+    .is_some()
+    {
+        let count: i64 = sqlx::query_scalar("SELECT count(*) FROM database_versions")
```
Rather than just checking the count, it may be better to check that `SELECT max(version) == 39` (and probably error in the case that the table exists but the version number _isn't_ that).

syphar replied:
valid point, done

syphar commented:
> > I removed running the reverse migrations from the teardown for every test case, I assume it's good enough to clean the whole schema. I added a separate check in CI to test the full setup & teardown only once.
> 
> Running the down migrations after each test was added to try and get a variety of data in the database that might cause issues in the down migration logic. Otherwise we're only really checking the down migration syntax, not what it does to data.

Thanks for the pointer, I added it back.
